ci-operator/step-registry/cucushift/installer/check/azure/identity/cucushift-installer-check-azure-identity-commands.sh

```diff
@@ -112,59 +112,97 @@ check_result=0
+            name=$(yq-go r ${INSTALL_CONFIG} "controlPlane.platform.azure.identity.userAssignedIdentities[$num].name")
+            rg=$(yq-go r ${INSTALL_CONFIG} "controlPlane.platform.azure.identity.userAssignedIdentities[$num].resourceGroup")
+            expected_identity_id_master+=" /subscriptions/${subscrption}/resourceGroups/${rg}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${name}"
+        done
+    fi
 fi
-identity_type_compute=$(yq-go r ${INSTALL_CONFIG} 'compute[0].platform.azure.identity.type')
-if [[ "${identity_type_compute}" == "UserAssigned" ]]; then
-    end_number=$((AZURE_USER_ASSIGNED_IDENTITY_NUMBER - 1))
-    compute_identity_id=""
-    for num in $(seq 0 ${end_number}); do
-        subscrption=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].subscription")
-        name=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].name")
-        rg=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].resourceGroup")
-        compute_identity_id+=" /subscriptions/${subscrption}/resourceGroups/${rg}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${name}"
-    done
+
+if [[ -n "${identity_type_worker}" ]]; then
+    expected_identity_type_worker="${identity_type_worker}"
+    if [[ "${identity_type_worker}" == "UserAssigned" ]] && [[ ${AZURE_USER_ASSIGNED_IDENTITY_NUMBER} -gt 0 ]]; then
+        expected_identity_id_worker=""
+        for num in $(seq 0 ${end_number}); do
+            subscrption=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].subscription")
+            name=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].name")
+            rg=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].resourceGroup")
+            expected_identity_id_worker+=" /subscriptions/${subscrption}/resourceGroups/${rg}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${name}"
+        done
+    fi
 fi
 
-echo "control_plane_identity_id: ${control_plane_identity_id}"
-echo "compute_identity_id: ${compute_identity_id}"
+echo "expected_identity_type_master: ${expected_identity_type_master}"
+echo "expected_identity_id_master: ${expected_identity_id_master}"
+echo "expected_identity_type_worker: ${expected_identity_type_worker}"
+echo "expected_identity_id_worker: ${expected_identity_id_worker}"
+echo "cluster_identity_id: ${cluster_identity_id}"
+echo "AZURE_USER_ASSIGNED_IDENTITY_NUMBER: ${AZURE_USER_ASSIGNED_IDENTITY_NUMBER}"
+
+if [[ "${expected_identity_type_master}" == "UserAssigned" ]] && [[ -z "${expected_identity_id_master}" ]]; then
+    echo "control plane's identity type is UserAssigned, but expected identity ids are empty, please check..."
```

Comment:
I guess it is allowed to set controlPlane.platform.azure.identity.type=UserAssigned but without any customer identity, right ?

Reply:
And per my understanding, if not configure anything about identity, the default type is actually `UserAssigned`, but the only difference is the identity is created by installer, but provided by users. That is why in my proposed codes:
```
if [[ -z "${identity_type_default}" ]]; then
    expected_identity_type_default="UserAssigned"
fi
expected_identity_type_control_plane="${expected_identity_type_default}"
expected_identity_type_compute="${expected_identity_type_default}"
```

Reply:
The default `expected_identity_id_master` is initialized to  to `cluster_identity_id`, then if user defines customer identity in install-config, `expected_identity_id_master` is overridden with customer identity.

Finally,  `expected_identity_id_master` should be have value if master identity type is set to `UserAssigned`.

Reply:
In the following cases:
```
platform.azure.defaultMachinePlatform.identity.type is not configured, 
controlPlane.platform.azure.identity.type is not configured, 
compute[0].platform.azure.identity.type=UserAssigned with customer identity
````
installer would still create an identity for masters, if it is true, sounds like we did not check if the identity created or not.

Reply:
right, updated as below, it should cover this check.
```
if ([[ -z "${expected_identity_type_master}" ]] || [[ -z "${expected_identity_type_worker}" ]]) && [[ -z "${cluster_identity_id}" ]]; then
    echo "No identity is specified on master or worker, installer should create an identity by default, but not found it, exit..."
    exit 1
fi
```